src/unix/text.c

```diff
@@ -42,18 +48,42 @@ value caml_gr_set_text_size (value sz)
 
 static void caml_gr_draw_text(const char *txt, int len)
 {
+  int rgb;
+  XftColor xftcol;
+  XftDraw *d;
+  Visual *visual = DefaultVisual (caml_gr_display, caml_gr_screen);
+  XGlyphInfo info;
+
+  rgb = caml_gr_rgb_pixel (caml_gr_color);
+  xftcol.pixel = rgb;
+  /* The range of these fields is 0..0xffff */
+  xftcol.color.red = rgb >> 8;;
+  xftcol.color.green = (rgb & 0xff00);
+  xftcol.color.blue = (rgb & 0xff) << 8;
+  xftcol.color.alpha = 0xffff;
+
   if (caml_gr_font == NULL) caml_gr_get_font(DEFAULT_FONT);
-  if (caml_gr_remember_modeflag)
-    XDrawString(caml_gr_display, caml_gr_bstore.win, caml_gr_bstore.gc,
-                caml_gr_x, Bcvt(caml_gr_y) - caml_gr_font->descent + 1, txt,
-                len);
+  if (caml_gr_remember_modeflag) {
+    d = XftDrawCreate (caml_gr_display, caml_gr_bstore.win,
+          visual, caml_gr_colormap);
+    XftDrawString8 (d, &xftcol, caml_gr_font,
+        caml_gr_x, Bcvt(caml_gr_y) - caml_gr_font->descent + 1,
+        (const FcChar8 *) txt, len);
+  }
   if (caml_gr_display_modeflag) {
-    XDrawString(caml_gr_display, caml_gr_window.win, caml_gr_window.gc,
-                caml_gr_x, Wcvt(caml_gr_y) - caml_gr_font->descent + 1, txt,
-                len);
+    d = XftDrawCreate (caml_gr_display, caml_gr_window.win,
+          visual, caml_gr_colormap);
+    XftDrawString8 (d, &xftcol, caml_gr_font,
+        caml_gr_x, Wcvt(caml_gr_y) - caml_gr_font->descent + 1,
+        (const FcChar8 *) txt, len);
     XFlush(caml_gr_display);
   }
-  caml_gr_x += XTextWidth(caml_gr_font, txt, len);
+
+  if (d) XftDrawDestroy (d);
```

Comment:
This destroys one XftDraw object but two may have been created.  Also, `d` is not initialized, if I'm not mistaken, so the `if (d)` test is weird.